src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -570,6 +732,43 @@ def _transform_single_condition(
     return condition, None
 
 
+def _get_filters_for_preflight_query_condition(
+    tag_name: str, condition: Union[Condition, BooleanCondition]
+) -> Tuple[Optional[Op], Optional[Set[str]]]:
+    """
+    Function that takes a tag name and a condition, and checks if that condition is for that tag
+    and if so returns a tuple of the op applied either Op.IN or Op.NOT_IN and a set of the tag
+    values
+    """
+    if isinstance(condition, Condition):
+        if condition.lhs == Column(tag_name):
+            if condition.op in [Op.EQ, Op.NEQ, Op.IN, Op.NOT_IN]:
```

Comment:
for readability better to combine it with and IMO:
```
if isinstance(condition, Condition) and condition.lhs == Column(tag_name) and condition.op in [Op.EQ, Op.NEQ, Op.IN, Op.NOT_IN]:
```
python won't evaluate second condition if first is False.

Reply:
I don't agree here .. I think its actually harder to read a line with all three conditions on it

Reply:
you can always split ands into 3 lines, but i'll leave it for you. I reread it carefully and we can't combine 3 ifs because if you have `raise InvalidParams` .... So best we can do here:
```diff
diff --git src/sentry/release_health/metrics_sessions_v2.py src/sentry/release_health/metrics_sessions_v2.py
index 5f327a6b73..0a7d0fb9d9 100644
--- src/sentry/release_health/metrics_sessions_v2.py
+++ src/sentry/release_health/metrics_sessions_v2.py
@@ -740,21 +740,18 @@ def _get_filters_for_preflight_query_condition(
     and if so returns a tuple of the op applied either Op.IN or Op.NOT_IN and a set of the tag
     values
     """
-    if isinstance(condition, Condition):
-        if condition.lhs == Column(tag_name):
-            if condition.op in [Op.EQ, Op.NEQ, Op.IN, Op.NOT_IN]:
-                filters = (
-                    {condition.rhs}
-                    if isinstance(condition.rhs, str)
-                    else {elem for elem in condition.rhs}
-                )
-                op = {Op.EQ: Op.IN, Op.IN: Op.IN, Op.NEQ: Op.NOT_IN, Op.NOT_IN: Op.NOT_IN}[
-                    condition.op
-                ]
-                return op, filters
-            raise InvalidParams(
-                f"Unable to resolve {tag_name} filter due to unsupported op {condition.op}"
+    if isinstance(condition, Condition) and condition.lhs == Column(tag_name):
+        if condition.op in [Op.EQ, Op.NEQ, Op.IN, Op.NOT_IN]:
+            filters = (
+                {condition.rhs}
+                if isinstance(condition.rhs, str)
+                else {elem for elem in condition.rhs}
             )
+            op = {Op.EQ: Op.IN, Op.IN: Op.IN, Op.NEQ: Op.NOT_IN, Op.NOT_IN: Op.NOT_IN}[condition.op]
+            return op, filters
+        raise InvalidParams(
+            f"Unable to resolve {tag_name} filter due to unsupported op {condition.op}"
+        )
```

Reply:
Agreed